doc/arch/adr-010-learn-to-rank.md

```diff
@@ -0,0 +1,286 @@
+# Decision record: Learning To Rank
+
+**Date:** 2019-11-07
+
+The search team have decided to implement [Learning To Rank][].
+
+1. What is Learning To Rank?
+1. Motivation
+1. Implementation
+1. What success will look like
+
+[Learning To Rank]: https://en.wikipedia.org/wiki/Learning_to_rank
+
+## What is Learning To Rank?
+
+Learning To Rank (LTR) is the application of machine learning to create a
+ranking model that can be used to improve search relevance.
+
+This is a common approach for advanced search applications.
+
+The ranking of search results on websites are frequently determined by
+machine learning models. For example, Airbnb, Yelp, Wikipedia, Bloomberg,
+and commercial search engines.
+
+There is a good explanation on what LTR is here:
+
+- https://elasticsearch-learning-to-rank.readthedocs.io/en/latest/core-concepts.html
+- https://opensourceconnections.com/blog/2017/02/24/what-is-learning-to-rank/
+
+### How does Learning To Rank work?
+
+It's easiest to explain with an example:
+
+Say you searched for 'Harry Potter'. Search API might return the following results:
+
+1. HMRC Sign in
+2. Harry Potter World
+3. Pottery barn
+4. Harry Potter
+
+These are not good results.
+
+Our approach now (prior to implementing Learning To Rank) would be to
+manually rank these results, giving them a score of between 0-3 with
+0 being irrelevant and 3 being completely relevant:
+
+```
+query         rating    document
+harry potter  0         HMRC Sign in
+harry potter  2         Harry Potter World
+harry potter  0         Pottery barn
+harry potter  3         Harry Potter
+```
+
+These judgements are an expression of the ideal ordering.
+
+We can then use nDCG to tell us how good the results are.
+We might get a score of 0.3 for this result set (from 0 to 1).
+
+We could then investigate why the results are bad. Perhaps there is a
+best bet. Or popularity might be pushing HMRC docs. We can fix these
+things, then re-run nDCG, and we might get 0.7 for this query.
+
+In contrast, with Learning To Rank, search result ranks would be
+determined by a model without manually changing the boost of a field.
+
+With Learning To Rank we would go through the same process of
+rating the search results.
+
+```
+harry potter  0         Pottery barn
+harry potter  3         Harry Potter
+...
+```
+
+However, after doing this, we would train a model with this data
+to rank the results better. The model optimises for nDCG.
+
+We provide the model with features, such as the fields we boost on.
```

Comment:
Add more explanation of how features are used:

> We train the model using using a combination of relevancy judgements and *features*, numeric measures which indicate how well we think the result matches the query. The text similarity scores Elasticsearch assigns to fields in the document are examples of features.  The model tries to find correlations between document features and relevancy judgements, which allows it to predict how to order documents we *don't* have relevancy judgements for.

Reply:
I see that this is explained a little further down, but I think it's good to put at least some of the explanation up front so the reader isn't left wondering what the point is.